### PR TITLE
Add docs for contributing

### PR DESCRIPTION
**Pull request by TheMarex**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Before you open a new issue, please search for older ones that cover the same issue.
+In general "me too" comments/issues are frowned upon.
+You can add a :+1: emoji to the issue if you want to express interest in this.
+
+# Developer
+
+We use `clang-format` version `3.8` to consistently format the code base. There is a helper script under `scripts/format.sh`.
+
+In general changes that affect the API and/or increase the memory consumption need to be discussed first.
+Often we don't include changes that would increase the memory consumption a lot if then are not generally usable (e.g. elevation data is a good example).
```
Comment (MoKob) [inline CONTRIBUTING.md]: if they

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Once your pull request is reviewed you can merge it! If you don't have commit access, ping someone that has commit access.
+If you do have commit access there are in general two accepted styles to merging:
+
+1. Make sure the branch is up to date with `master`. Run `git rebase master` to find out.
+2. Once that is ensure you can either:
+  - Click the nice green merge button
+  - Merge by hand using a fast-forward merge
+
+Which merge you prefer is up to personal preference. In general it is recommended to use fast-forward merges because it creates a history that is sequential and easier to understand.
```
Comment (daniel-j-h) [inline CONTRIBUTING.md]: make clear gh-button does not ff-merge

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+# Maintainer
+
+## Doing a release
+
+There is an in-depth guide around how to push out a release once it is ready [here](docs/releasing.md)
```
Comment (daniel-j-h) [inline CONTRIBUTING.md]: .

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## The API
+
+Changes to the API need to be discussed and signed off by the team. Breaking changes even more so than additive changes.
+
+## Milestones
+
+If a pull request or issue is applicable for the current next milestone, depends on the target version number.
+Since we use semantic versioning we restrict breaking changes to major releases.
+After an Release Candidate is released we usually don't change the API anymore if it is not critical.
```
Comment (daniel-j-h) [inline CONTRIBUTING.md]: a

**Pull request merged**